addons/sms/tests/test_sms_composer.py

```diff
@@ -0,0 +1,118 @@
+
+                self.assertSMSNotification(
+                    [{'number': '+3215228817386'}], expected_sms_content, message,
+                    mail_message_values={"body": expected_notification_content}
+                )
+
+    def test_message_sms_body_sms_vs_notification(self):
+        """Check that the rendering of the sms notification is identical to the sms.
+
+        The only expected difference is that links are converted to be clickable."""
+        # Cases are formatted as (sms text, expected notification body, old notification body rendering)
+        # The last element is used to show what bug is fixed with this commit from 15.0 and should be
+        # cleaned in master.
+        cases = [
+            (
+                "Hello there, check this awesome app I found:\nhttps://odoo.com",
+                '<p>Hello there, check this awesome app I found:<br>'
+                '<a href="https://odoo.com" target="_blank" rel="noreferrer noopener">https://odoo.com</a></p>',
+                # same
+                '<p>Hello there, check this awesome app I found:<br/>'
+                '<a href="https://odoo.com" target="_blank" rel="noreferrer noopener">https://odoo.com</a></p>',
+            ), (
+                "Hello there, check this awesome <b>app</b> I found:\nhttps://odoo.com",
+                # b is kept as is in notification, but link is still added as well
+                '<p>Hello there, check this awesome &lt;b&gt;app&lt;/b&gt; I found:<br>'
+                '<a href="https://odoo.com" target="_blank" rel="noreferrer noopener">https://odoo.com</a></p>',
+                # html was interpreted and converted
+                '<p>Hello there, check this awesome *app* I found:<br/>'
+                '<a href="https://odoo.com" target="_blank" rel="noreferrer noopener">https://odoo.com</a></p>',
+            ),
+        ]
+        for sms_content, expected_notification_content, old_expected_notification_content in cases:
+            with self.subTest(sms_content=sms_content):
+                # compare with old rendering
+                old_notification_content = plaintext2html(html2plaintext(sms_content))
+                self.assertEqual(old_notification_content, old_expected_notification_content)
+                with self.with_user('admin'):
+                    composer = self.env['sms.composer'].with_context(
+                        active_model='res.partner', active_id=self.partner_employee).create({'body': sms_content})
+
+                    with self.mockSMSGateway():
+                        messages = composer._action_send_sms()
+
+                self.assertSMSNotification(
+                    [{'partner': self.partner_employee}], sms_content, messages,
+                    mail_message_values={"body": expected_notification_content},
+                    from_text=True,
+                )
+
+    def test_message_sms_body_preserved(self):
```

Comment:
Can't we include it in previous test so that we test the composer behavior ? Would say it is clearer to have all composer tests with some different cases in it, isn't it ?